### ISSUE TITLE
Experiment C: Non-linear gradient-based projection

### ISSUE THREAD

**Issue by gelijergensen** [gelijergensen/Constrained-Neural-Nets-Workbook#101]
We came up with completely new idea which we are switching to with the remaining 12.5 days. Namely, we are going to train the model the normal, unconstrained way, and then at inference time, we use the constraint to back-propagate and (temporarily) tweak the neural network parameters to find the nearest point on the constraint surface. In other words, we are going to do non-linear gradient-based projection at inference-time, which will slow down inferencing, but it does make this method applicable to even pre-trained models.

Here, we want to simply produce a proof of concept for this on the sine data and show that it does work in reducing the constraints at inference time
